flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/sql/JoinITCase.scala

```diff
@@ -372,9 +372,163 @@ class JoinITCase(
     val table = CollectionDataSets.getSmall3TupleDataSet(env).toTable(tEnv).as('a1, 'a2, 'a3)
     tEnv.registerTable("A", table)
 
-    val sqlQuery1 = "SELECT * FROM A CROSS JOIN (SELECT count(*) FROM A HAVING count(*) < 0)"
-    val result = tEnv.sql(sqlQuery1).count()
+    val sqlQuery1 = "SELECT * FROM A CROSS JOIN " +
+      "(SELECT count(*) FROM A HAVING count(*) < 0)"
+    val result = tEnv.sql(sqlQuery1)
+    val expected =Seq(
+      "2,2,Hello,null",
+      "1,1,Hi,null",
+      "3,2,Hello world,null").mkString("\n")
 
-    Assert.assertEquals(0, result)
+    val results = result.toDataSet[Row].collect()
+    TestBaseUtils.compareResultAsText(results.asJava, expected)
+  }
+
+  @Test
+  def testSingleRowLeftOuterJoin(): Unit = {
+
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env, config)
+
+    val sqlQuery =
+      "SELECT  a, cnt FROM t1 " +
+        "LEFT JOIN (" +
+        "SELECT COUNT(*) AS cnt FROM t2" +
+        ") AS x " +
+        "ON a > cnt"
+
+    val ds1 = CollectionDataSets.get5TupleDataSet(env).toTable(tEnv).as('a, 'b, 'c, 'd, 'e)
+    val ds2 = CollectionDataSets.getSmall3TupleDataSet(env).toTable(tEnv)
+    tEnv.registerTable("t1", ds1)
+    tEnv.registerTable("t2", ds2)
+
+    val result = tEnv.sql(sqlQuery)
+
+    val expected = Seq(
+      "1,null",
+      "2,null", "2,null",
+      "3,null", "3,null", "3,null",
+      "4,3", "4,3", "4,3", "4,3",
+      "5,3", "5,3", "5,3", "5,3", "5,3").mkString("\n")
+
+    val results = result.toDataSet[Row].collect()
+    TestBaseUtils.compareResultAsText(results.asJava, expected)
+  }
+
+  @Test
+  def testSingleRowRightOuterJoin(): Unit = {
```

Comment:
This case don't pass. Because ``right join`` in this case execute via ``MapJoinLeftRunner``.  please see comment in [jira](https://issues.apache.org/jira/browse/FLINK-5256?focusedCommentId=15825599&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15825599)